s3bench.py

```diff
@@ -162,29 +175,22 @@ def list_random_objects(self):
         """This function returns randomized list of object in a bucket according to a given number
         In case number is bigger than 1000, use pagination, else use regular v2"""
         # in case number of objects is smaller then the page size, to save list costs
-        if int(self.num_objects) <= 1000:
-            objects = self.s3.list_objects(Bucket=self.bucket_name, MaxKeys=int(self.num_objects))
-        else:
-            keys = []
-            # uses pagination to list object number bigger then 1000
-            paginator = self.s3.get_paginator('list_objects')
-            pages = paginator.paginate(Bucket=self.bucket_name)
-            for page in pages:
-                for obj in page['Contents']:
-                    keys.append({'Key':obj['Key'], 'Size':obj['Size']})
-            # shuffles keys list randomly to create different list between clients
-            random.shuffle(keys)
-            # picks a random object from the list
-            random_object_index = keys.index(random.choice(keys))
-            # creates a circular list for list index deviation
-            circular_keys = keys + keys
-            # returns a slice of keys list according to num_objects var
-            return circular_keys[random_object_index:random_object_index + int(self.num_objects)]
-        # in case there is no need for pagination and there are objects to read in the bucket
-        if 'Contents' in objects:
-            return objects['Contents']
-        else:
-            raise Exception("no objects to read in bucket!")
+        keys = []
+        # uses pagination to list object number bigger then 1000
+        paginator = self.s3.get_paginator('list_objects')
+        pages = paginator.paginate(Bucket=self.bucket_name, Prefix=self.object_size)
```

Comment:
1. Here we should be more generic to guarantee the even if the customer doesn't use this option of `--prefix` we still list the whole bucket. 

Why? 
1. The current method is not generic and assumes that the prefix is always the object name, we should provide the ability of PUTting and GETting from any given prefix.

```suggestion
pages = paginator.paginate(Bucket=self.bucket_name, Prefix=self.prefix)
```

Reply:
we therefore still leave it to the client to make sure read workloads actually list on objects of the specified size?

there is no set correlation between the prefix and the object size, so the pr name should probably change in accordance to these change.

Reply:
Agreed. Let's maybe have a new PR with all the suggested fixes and then we'll merge it. What do you think?